jsonschema-core.xml

```diff
@@ -125,57 +125,51 @@
             </t>
         </section>
 
-        <section title="Overview">
+        <!-- JSON Schema accomplishes two objectives, which each get their own section. -->
+        <section title="Validation">
             <t>
-                This document proposes a new media type "application/schema+json" to identify a JSON
-                Schema for describing JSON data.
-                It also proposes a further optional media type, "application/schema-instance+json",
-                to provide additional integration features.
-                JSON Schemas are themselves JSON documents.
-                This, and related specifications, define keywords allowing authors to describe JSON
-                data in several ways.
+                A JSON Schema document describes a validator (also known as a "recognizer" or "acceptor") which classifies a provided JSON document as "accepted" or "rejected."
```

Comment:
Does the schema describe a validator?  I would expect people think of the "validator" as the implementation, not the document.

Reply:
Yeah that makes sense... There's a sense in which these two uses are actually the same, the "validator implementation" is just a generic form of validator that is configurable. Like if I have a schema, then if the program is written or compiled to work only with that schema, or if it's generic and configured at runtime, makes no difference.

Is there a better name for "the program that tests an input against some specific schema"?

Reply:
I don't think you understand. Colloquially, the "validator" _is_ the implementation, not the schema.  I think we need to stick with this.

Saying the schema itself is the validator will be confusing.  A validator evaluates JSON _against_ a schema.  The schema is no more than configuration.

Reply:
> Colloquially, the "validator" is the implementation, not the schema.

I believe I see the point you're making, but I'm adding, this is similar to how we discuss compilers and interpreters. You're pointing out a definition of "validator" that functions like an interpreter: there's a library that reads the schema (the source code), then uses this interpretation to validate JSON.

But you can also compile source code to a program, and run the program directly. In this paradigm, there is no interpreter (what is usually called the validator), but the compiled program is still a "validator" (a thing that performs validation). It just has no concept of a schema (any more than a compiled C program can parse C).

So with JSON Schema, the schema is not the validator (as such), but I think you can say it _describes_ a validator.

Reply:
I see where you're coming from, but never in my experience with this project have we used "validator" that way.  It has always been used to mean the implementation.

Reply:
@gregsdennis I think the interpreter/compiler paradigm is a useful way to think about it because it anticipates a broader range of use cases... but I can understand how it sounds weird, do you have any suggestions?

I don't think this is actually incompatible with how we've been using the term, would additional explanation help?

Reply:
I think at this point we should get other opinions.

Reply:
@gregsdennis Well, what's your opinion? Specifically, if I write a program that's not an interperter (like most validators), but a compiler (i.e. the output is a single Javascript function that takes one argument, the instance)—what term should be used to refer to that _compiled_ function?

@Julian I'd like to hear your feedback too.

Reply:
I don't believe we need terminology for such a concept in the spec at all (and certainly not at this point in time). What we use today is fine, "implementation", which refers to the executable program capable of doing things with schemas.

Reply:
> the function that accepts the instance would be the "validator"

This I agree with, but it doesn't follow from this that the _schema_ is a validator.  The schema is still just "configuration" (if you want to call it that.  It still goes through a library/application, and you get an output.  It's just that your example also produces an intermediate output of an executable function that represents a specific schema.  The _system_ is inputting the JSON Schema (most likely as JSON or YAML text) and an instance and getting out whether the instance is valid according to that schema.  That "compile" step is an intermediate implementation detail that doesn't need to be covered in the spec.

The spec needs to concern itself with one thing:

- inputs: a schema and an instance
- output: validation results and/or annotations

Anything an implementation does to get from input to output is necessarily beyond the scope of the spec.

Reply:
> but it doesn't follow from this that the schema is a validator

I see, this isn't what I intended to convey. By saying "the schema _describes_ a validator" I think that would disconnect the schema (the description) from the validator (the actual process). Is a different word is in order here, or some additional explanation ("the schema describes the behavior of a validator")?